rdt/transformers/numerical.py

```diff
@@ -487,3 +490,154 @@ def _reverse_transform(self, data):
             data = self._univariate.ppf(scipy.stats.norm.cdf(data))
 
         return super()._reverse_transform(data)
+
+
+class BayesGMMTransformer(NumericalTransformer):
+    """Bayesian GMM transformer."""
+
+    STD_MULTIPLIER = 4
+    DETERMINISTIC_TRANSFORM = False
+    DETERMINISTIC_REVERSE = True
+    COMPOSITION_IS_IDENTITY = False
+
+    def __init__(self, dtype=None, nan='mean', null_column=None, max_clusters=10,
+                 weight_threshold=0.005):
+        super().__init__(dtype=dtype, nan=nan, null_column=null_column)
+        self._max_clusters = max_clusters
+        self._weight_threshold = weight_threshold
+        self._number_of_modes = None
+        self._bgm_transformer = None
+        self._valid_component_indicator = None
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        output_types = {
+            'continuous': 'float',
+            'discrete': 'categorical'
+        }
+        if self.null_transformer and self.null_transformer.creates_null_column():
+            output_types['is_null'] = 'float'
+
+        return self._add_prefix(output_types)
+
+    def _fit(self, data):
+        """Fit the transformer to the data.
+
+        Args:
+            data (pandas.Series):
+                Data to fit to.
+        """
+        self._bgm_transformer = BayesianGaussianMixture(
+            n_components=self._max_clusters,
+            weight_concentration_prior_type='dirichlet_process',
+            weight_concentration_prior=0.001,
+            n_init=1
+        )
+
+        super()._fit(data)
+        data = super()._transform(data)
+        if data.ndim > 1:
+            data = data[:, 0]
+
+        self._bgm_transformer.fit(data.reshape(-1, 1))
+        self._valid_component_indicator = self._bgm_transformer.weights_ > self._weight_threshold
+        self._number_of_modes = self._valid_component_indicator.sum()
+
+    def _transform(self, data):
+        """Transform numerical data.
+
+        Args:
+            data (pandas.Series):
+                Data to transform.
+
+        Returns:
+            numpy.ndarray
+        """
+        data = super()._transform(data)
+        if data.ndim > 1:
```

Comment:
what is the scenario where this is true? If this is because of the null column, are we guaranteed to get the columns in the correct order?

Reply:
It's because of the NullTransformer. It should always be the last column, since in the NullTransformer code we concatenate the null_column at the end:

https://github.com/sdv-dev/RDT/blob/b7e7f213044b067d6bcd258baffb6c67c26f9328/rdt/transformers/null.py#L125-L126